src/plugins/intel_gpu/src/runtime/kernels_cache.cpp

```diff
@@ -327,12 +344,22 @@ void kernels_cache::build_batch(const engine& build_engine, const batch_program&
 
             program.createKernels(&kernels);
 
+            auto program_bin = getProgramBinaries(program);
+
+            // save to in_memory cache
+            // Cache is saved at both in-memory and file, because in-memory cache capacity is limited.
+            if (batch.kernels_counter == 1) {
```

Comment:
This is not that informative. I think we'll need some option whether to use in memory cache expected in the future. 
Though we dont have concrete conclusion on the options, lets make this as a function which returns whether in memory cache is turned on or not.